File: VehicleChargingSimulationFE/src/Controllers/Message.java
package Controllers;

import Entities.Scenario;

public class Message {
	
	enum Operation{
		RUN,
		EXPORT,
		IMPORT
	}
	
	private final int _version = 1;
	private Operation _operation;
	private Scenario _scenario;
	
	Message(){}
	
	Message(Operation operation, Scenario scenario){
		_operation = operation;
		_scenario = scenario;
	}
	
	@Override
	public String toString() {
		return _version + _operation.name() + _scenario.toString();
	}
}

File: VehicleChargingSimulationScripts/entities/Message.py
from enum import Enum

from entities.Scenario import Scenario


class Message:
    class Operation(Enum):
        RUN = 0
        EXPORT = 1
        IMPORT = 2

    def __init__(self):
        self.scenario = None
        self.version = 1
        self.operation = -1

    @classmethod
    def parseMessage(cls, string):
        message = Message()

        splitString = string.split(";")
        versionOperation = splitString[1].split("operation=")

        versionOperation[1].strip()

        if versionOperation[1] == "RUN":
            message.operation = 0
        elif versionOperation[1] == "EXPORT":
            message.operation = 1
        elif versionOperation[1] == "IMPORT":
            message.operation = 2
        message.scenario = Scenario.parseScenario(splitString[2])
        return message
